test/http-mock-server.h

```diff
@@ -306,7 +311,7 @@ class HttpMockServer {
   }
 
   std::string host() const {
-    return endpoint_.address().to_string();
+    return "localhost";
```

Comment:
And again let's add a comment that refers to the comment above on why we need to use the loopback address and thus `localhost` here. Thanks!

Reply:
Added!